### PR TITLE
Gradient background

### PR DESCRIPTION
**Pull request by sudo-ulmas**
[//]: # (Feel free to remove parts that do not relate to the changes you are willing to bring into main)

### Issue / Motivation
[//]: # (Give a brief summary of why the change is being implemented, i.e. feature was built, bug fixing and etc.)

### Add background gradient to all views
[//]: # (List change details here for Code Reviewers)
1. add gradient to table views and make cells background transparent
2. add gradient to detailed views
3. add gradient to scroll views 

[//]: # (Use the table below if you have screenshots/videos to show visual updates)
Simulator Screen Shot - iPhone 13 mini - 2023-01-12 at 14 59 24

### CODE REVIEW

File: SteamList/Views/GradientView.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    
+    override class var layerClass: AnyClass {
+        get {
+            return CAGradientLayer.self
+        }
+    }
+    
+    func updateView() {
+        let layer = self.layer as! CAGradientLayer
```
Comment (abubakrj) [inline SteamList/Views/GradientView.swift]: Not sure wether there will be any cases when layer is not an instance of `CAGradientLayer` class, but it is still good to wrap this via guard statement

File: SteamList/Views/GradientView.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    }
+    
+    override class var layerClass: AnyClass {
+        get {
+            return CAGradientLayer.self
+        }
+    }
+    
+    func updateView() {
```
Comment (abubakrj) [inline SteamList/Views/GradientView.swift]: This method is only being called to update the layer property. Better to ranme it to `updateLayer` or `updateGradientColors` etc

**Review by abubakrj — APPROVED** [review]
Good job, left a minor comment

File: SteamList/Views/GradientView.swift (hunk 1 above)
Comment (sudo-ulmas) [reply SteamList/Views/GradientView.swift]: done

File: SteamList/Views/GradientView.swift (hunk 2 above)
Comment (sudo-ulmas) [reply SteamList/Views/GradientView.swift]: done

**Pull request merged**